pyquil/wavefunction.py

```diff
@@ -145,28 +146,43 @@ def pretty_print_probabilities(self, decimal_digits=2):
                 outcome_dict[outcome] = prob
         return outcome_dict
 
-    def pretty_print(self, decimal_digits=2):
+    def pretty_print(self, decimal_digits=2, ancillas: Optional[List[int]] = None):
         """
         Returns a string repr of the wavefunction, ignoring all outcomes with approximately zero
         amplitude (up to a certain number of decimal digits) and rounding the amplitudes to
         decimal_digits.
 
         :param int decimal_digits: The number of digits to truncate to.
+        :param List[int] ancillas: The list of indices for ancillary qubits to omit.
         :return: A dict with outcomes as keys and complex amplitudes as values.
         :rtype: str
         """
+
+        if ancillas is None:
+            ancillas = []
+
+        mask = 0
+        for ancilla in ancillas:
+            mask |= (1 << ancilla)
+
         outcome_dict = {}
         qubit_num = len(self)
         pp_string = ""
         for index, amplitude in enumerate(self.amplitudes):
+            if index & mask != 0 and not np.isclose(amplitude, 0):
+                raise ValueError('Specified ancilla was found to have non-zero amplitude.')
             outcome = get_bitstring_from_index(index, qubit_num)
+            for ancilla in ancillas:
+                outcome = outcome[0:qubit_num - ancilla - 1] + outcome[qubit_num - ancilla:]
```

Comment:
i find this a bit hard to reason about -- add a comment?

Reply:
also, im wondering if we could do this by just grabbing the amplitudes we care about rather than editing each outcome? if we build a new amplitudes array up front then the whole outer `for` loop stays the same. that seems easier to reason about for me at least, but i'm happy to have a discussion

Reply:
I'm fine with either, my only objection would be that this is usually more efficient (and I personally find it more intuitive) to remove from the existing `outcome` than to build up a new one from scratch and omit the ancillae. Your call though- what do you think?